SCClassLibrary/Common/Collections/SequenceableCollection.sc

```diff
@@ -111,9 +111,16 @@ SequenceableCollection : Collection {
 	asSequenceableCollection { ^this }
 
 	// select an element at random
-	choose {
-		^this.at(this.size.rand)
+	choose { arg weights=nil;
+		if (weights.isNil) {
+			// with equal probabilities
+			^this.at(this.size.rand)
```

Comment:
you could factor out the `^` before the `if`. Then it is clear that we return something out of the method in any case.